src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1682,7 +1688,27 @@ static bool processSubscriptions
     }
 
     /* Check 2: expression (q) */
-    if (!matchExpression(notifyCerP, trigs->expression.q))
+    if (noCache == true)
```

Comment:
Not sure if I understand this correctly... It seems that TriggeredSubscription uses both `expression` (with string) and `stringFilterP` (parsed object). The later is used directly in the cache case, the former is used in the no-cache case, with an "on the fly" expression -> StringFilter conversion step. Is that correct?

Why don't simply TriggeredSubscription (and the logic in this part of the code) removing the `expression` field and creating the objects pinted by stringFilterP at TriggeringSubscription construction/setting time (if I'm remembering correctly, in the addTrriggeredSubscptions() function).